### PR TITLE
Make store tags optional

### LINKED ISSUE #2407: Make store_tags optional (sync integration)
Program settings storeTags in mSupply can't be empty:

image

Also see:
https://github.com/openmsupply/msupply/issues/13664#issuecomment-1765607162

For programs there might be no `storeTags` though. However, omSupply assumes that storeTags are not null. Thus integration fails when storeTags is null OR mSupply crashes if storeTags are empty.

Probably mSupply could handle this without crashing. However, we can also catch it in omSupply and make storeTags optional (LegacyStoreTagAndProgramName)

### PR DESCRIPTION
**Pull request by clemens-msupply**
Fixes #2407 

# 👩🏻‍💻 What does this PR do? 
Make the storeTags optional. Reason is that mSupply can't handle and empty storeTags object right now.

# Testing run integration tests

### CODE REVIEW

File: server/service/src/sync/translations/program_requisition_settings.rs [hunk 1]
```diff
@@ -27,14 +27,7 @@ pub struct LegacyListMasterRow {
...
 struct LegacyProgramSettings {
     #[serde(rename = "storeTags")]
-    store_tags: LegacyStoreTagAndProgramName,
-}
-
-#[derive(Deserialize, Clone)]
-struct LegacyStoreTagAndProgramName {
-    // HashMap key is the program name
-    #[serde(flatten)]
-    tags: HashMap<String, LegacyProgramSettingsStoreTag>,
```
Comment (clemens-msupply) [inline server/service/src/sync/translations/program_requisition_settings.rs:37]: @roxy-dao was there any reason to use flatten? and this struct?

Comment (roxy-dao) [reply server/service/src/sync/translations/program_requisition_settings.rs:37]: Not one I can remember anymore XD

Comment (clemens-msupply) [reply server/service/src/sync/translations/program_requisition_settings.rs:37]: I find the new version simpler, so if you don't object I would use this now...

**Review by mark-prins — APPROVED** [review]
👍 LGTM! 
disclaimer: I didn't test.. but I trust that you have. and the changes look fine to me.

**Pull request merged**